docs/tools/mbed_cli_2/build-system/profiles.md

```diff
@@ -1 +1,46 @@
-# Profiles
+# Mbed OS build profile
+
+The build profile defines the set of flags that is passed to the underlying compiler suite. In cmake based build system, `CMAKE_BUILD_TYPE` usually determines the build profile and normally this type is set by application target CMake to select the profile.
+
+## User-defined build profile
+
+In Mbed OS CMake based build system has dedicated cmake profile scripts for every build profiles:
+
+1. [develop.cmake](https://github.com/ARMmbed/mbed-os/blob/master/tools/cmake/profiles/develop.cmake)
+2. [debug.cmake](https://github.com/ARMmbed/mbed-os/blob/master/tools/cmake/profiles/debug.cmake)
+3. [release.cmake](https://github.com/ARMmbed/mbed-os/blob/master/tools/cmake/profiles/release.cmake)
+
+The build system includes one of the profile scripts into the build depending on CMAKE_BUILD_TYPE. The CMAKE_BUILD_TYPE is set by Mbed CLI 2 based on the argument it receives in the "-b" or "--profile" command line. If Mbed CLI 2 does not receive any argument, then it sets CMAKE_BUILD_TYPE to "Develop" based on that build system includes the corresponding "develop.cmake" script into the build.
+
+### Example
+
+Select `debug` profile using Mbed CLI 2 command:
+
+```
+ mbedtools compile -t GCC_ARM -m k64f -b debug
+ ```
+
+In case the application is using the direct CMake command then selecting the build type by passing `-DCMAKE_BUILD_TYPE` argument and if there is no `CMAKE_BUILD_TYPE` argument passed with cmake configuration command then the build system sets `CMAKE_BUILD_TYPE` to `develop`.
+
+### Example
+
+Select `release` profile using direct CMake command:
+
+```
+cmake -GNinja .. -DCMAKE_BUILD_TYPE=release
+```
+
+## Build profile script format
+
+Every profile script has the function `mbed_set_profile_options` which expects target and toolchain as an argument. This function is setting the necessary flags to the target using generator expression and target link options according to the toolchain argument. This function gets [called](https://github.com/ARMmbed/mbed-os/blob/master/CMakeLists.txt#L57) from Mbed OS root CMake.
```

Comment:
> This function gets [called](https://github.com/ARMmbed/mbed-os/blob/master/CMakeLists.txt#L57) from Mbed OS root CMake.

I would not be that specific (CMakeLists) is subject to change. It's easy to find where it is invoked.